### ISSUE TITLE
False warning: resource reference with title with capital letter

### ISSUE THREAD

**Issue by deric** [voxpupuli/puppet-lint-resource_reference_syntax#4]
Incorrect warning yield in case that type contains just variable:
```puppet
File[$conf_file],
```

**Comment by tuxmea** [comment]
Can you please verify which version of this lint plugin you use?
This one was already solved in October 2015.

**Comment by deric** [comment]
`puppet-lint-resource_reference_syntax (1.0.6)`

**Comment by tuxmea** [comment]
```
class demo::config {
  $conf_file = '/tmp/bar'
  $conf_file2 = '/tmp/bar2'
  file { $conf_file2:
    ensure    => file,
    subscribe => File[$conf_file],
  }
}
```
Work as expected. No error shows up.
Can you please provide your complete code example?

**Comment by deric** [comment]
```puppet
  file { '/etc/beegfs/beegfs-client.conf':
    ensure  => present,
    require =>[
      Package['beegfs-utils'],
      File[$interfaces_file]
    ],
  }
```

**Comment by tuxmea** [comment]
confirmed. same pattern as the other issue.

**Comment by tuxmea** [comment]
Can you please verify whether 1.0.7 fixes the issue?

**Comment by deric** [comment]
Partially fixed, not so many errors:
```puppet
 file { '/etc/beegfs/beegfs-meta.conf':
    ensure  => present,
    owner   => $user,
    group   => $group,
    mode    => '0755',
    content => template("beegfs/${major_version}/beegfs-meta.conf.erb"),
    require => [
      File[$interfaces_file], # ERROR: resource reference with title with capital letter on line 41
      Package['beegfs-meta'],
    ],
  }

  service { 'beegfs-meta':
    ensure     => running,
    enable     => $enable,
    hasstatus  => true,
    hasrestart => true,
    require    => Package['beegfs-meta'],
    subscribe  => [
      File['/etc/beegfs/beegfs-meta.conf'], # no error here
      File[$interfaces_file]
    ],
  }
```

**Comment by tuxmea** [comment]
Many thanks for reporting the issues.
The second one should be fixed with version 1.0.8 which is already uploaded to rubygems.org.
